addons/web/static/src/js/views/form_common.js

```diff
@@ -791,8 +791,22 @@ var AbstractField = FormWidget.extend(FieldInterface, {
     },
     on_translate: function() {
         var self = this;
+        function determineModel () {
+            var model = {};
+            model.name = self.view.model;
+            model.id = self.view.datarecord.id;
+            if (self.field.related) {
+                var fieldRelation = self.field.related[0];
+                model.name = self.field_manager.fields[fieldRelation].field.relation;
+                model.id = self.view.datarecord[fieldRelation];
+            }
+            return model;
+        }
+
+        var model = determineModel();
+
```

Comment:
func name is weird, but a function is actually not necessary:
```
var model = this.view.model;
var resID = this.view.datarecord.id;
if (this.field.related) {
    var fieldRelation = this.field.related[0];
    model = this.field_manager.fields[fieldRelation].field.relation;
    resID = this.view.datarecord[fieldRelation];
}

var trans = new data.DataSet(this, 'ir.translation');
return trans.call_button('translate_fields', [model, resID, this.name, this.view.dataset.get_context()]).done(function(r) {
    self.do_action(r);
});
```

Reply:
@aab-odoo 
Indeed